### PR TITLE
[#3574] Fix wrong context used in PubSubPublisherClient.publish().

### LINKED ISSUE #3574: Issues with MQTT devices subscribing and unsubscribing to the command topic
We are having issues with MQTT devices subscribing and unsubscribing to the command topic. Somehow a lot of them are timing out after 5 seconds of trying to register or unregister the consumer with the command router service.

Example log:
```
2023-10-24 15:33:27,025 INFO  [org.ecl.hon.cli.com.ProtocolAdapterCommandConsumerFactoryImpl] (vert.x-eventloop-thread-0) error registering consumer with the command router service [tenant: some-tenant, device: some-device, sendEvent: true]: org.eclipse.hono.client.ServerErrorException: request timed out after 5000ms
	at org.eclipse.hono.client.amqp.RequestResponseClient.lambda$sendRequest$6(RequestResponseClient.java:707)
	at org.eclipse.hono.client.amqp.RequestResponseClient.lambda$cancelRequest$3(RequestResponseClient.java:411)
	at java.base/java.util.Optional.map(Unknown Source)
	at org.eclipse.hono.client.amqp.RequestResponseClient.cancelRequest(RequestResponseClient.java:410)
	at org.eclipse.hono.client.amqp.RequestResponseClient.lambda$sendRequest$7(RequestResponseClient.java:707)
	at io.vertx.core.impl.VertxImpl$InternalTimerHandler.handle(VertxImpl.java:932)
	at io.vertx.core.impl.VertxImpl$InternalTimerHandler.handle(VertxImpl.java:903)
	at io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:55)
	at io.vertx.core.impl.ContextBase.emit(ContextBase.java:239)
	at io.vertx.core.impl.ContextInternal.emit(ContextInternal.java:194)
	at io.vertx.core.impl.VertxImpl$InternalTimerHandler.run(VertxImpl.java:921)
	at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)
	at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:153)
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:406)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Unknown Source)
```
Sadly the spans also don't really help us as they look like this:

image
The second log was added by me so it can be disregarded.

The only thing to go on really are some strange exceptions in the command router logs.

**A lot of ClassCastExceptions without a stack trace:**

image

Usually they were all uncaught exceptions but I added an exception handler to the context during the start() method

image

**Also some IllegalArgumentExceptions:**
Some of them are still uncaught:

```
2023-10-24 15:24:52,161 ERROR [io.qua.ver.cor.run.VertxCoreRecorder] (vert.x-eventloop-thread-1) Uncaught exception received by Vert.x: java.lang.IllegalArgumentException: newPosition > limit: (29 > 0)
	at java.base/java.nio.Buffer.createPositionException(Unknown Source)
	at java.base/java.nio.Buffer.position(Unknown Source)
	at java.base/java.nio.ByteBuffer.position(Unknown Source)
	at java.base/java.nio.ByteBuffer.position(Unknown Source)
	at io.netty.buffer.UnsafeByteBufUtil.setBytes(UnsafeByteBufUtil.java:563)
	at io.netty.buffer.PooledUnsafeDirectByteBuf.setBytes(PooledUnsafeDirectByteBuf.java:205)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1113)
	at io.vertx.proton.impl.ProtonTransport.flush(ProtonTransport.java:277)
	at io.vertx.proton.impl.ProtonConnectionImpl.flush(ProtonConnectionImpl.java:296)
	at io.vertx.proton.impl.ProtonReceiverImpl.flushConnection(ProtonReceiverImpl.java:175)
	at io.vertx.proton.impl.ProtonReceiverImpl.flow(ProtonReceiverImpl.java:149)
	at io.vertx.proton.impl.ProtonReceiverImpl.flow(ProtonReceiverImpl.java:135)
	at org.eclipse.hono.service.amqp.AbstractRequestResponseEndpoint.flowCreditToRequestor(AbstractRequestResponseEndpoint.java:519)
	at org.eclipse.hono.service.amqp.AbstractRequestResponseEndpoint.lambda$handleRequestMessage$10(AbstractRequestResponseEndpoint.java:363)
	at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Composition$1.onSuccess(Composition.java:62)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Mapping.onSuccess(Mapping.java:40)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Otherwise.onSuccess(Otherwise.java:28)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Composition$1.onSuccess(Composition.java:62)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.SucceededFuture.addListener(SucceededFuture.java:88)
	at io.vertx.core.impl.future.Composition.onSuccess(Composition.java:43)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Composition$1.onSuccess(Composition.java:62)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Composition$1.onSuccess(Composition.java:62)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.SucceededFuture.addListener(SucceededFuture.java:88)
	at io.vertx.core.impl.future.Composition.onSuccess(Composition.java:43)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Composition$1.onSuccess(Composition.java:62)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.SucceededFuture.addListener(SucceededFuture.java:88)
	at io.vertx.core.impl.future.Composition.onSuccess(Composition.java:43)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Mapping.onSuccess(Mapping.java:40)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Otherwise.onSuccess(Otherwise.java:28)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Mapping.onSuccess(Mapping.java:40)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Composition$1.onSuccess(Composition.java:62)
	at io.vertx.core.impl.future.FutureImpl$ListenerArray.onSuccess(FutureImpl.java:262)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.FixedMapping.onSuccess(FixedMapping.java:31)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Composition$1.onSuccess(Composition.java:62)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.SucceededFuture.addListener(SucceededFuture.java:88)
	at io.vertx.core.impl.future.Composition.onSuccess(Composition.java:43)
	at io.vertx.core.impl.future.FutureImpl$ListenerArray.onSuccess(FutureImpl.java:262)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23)
	at io.vertx.core.Promise.complete(Promise.java:66)
	at org.eclipse.hono.client.pubsub.publisher.PubSubPublisherClientImpl$1.lambda$onSuccess$0(PubSubPublisherClientImpl.java:128)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:264)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:246)
	at io.vertx.core.impl.EventLoopContext.lambda$runOnContext$0(EventLoopContext.java:43)
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:403)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Unknown Source)
```

Some of them are caught somewhere:

```
2023-10-24 14:22:50,507 WARN  [org.ecl.hon.com.app.Application$1] (vert.x-eventloop-thread-0) error handling message: java.lang.IllegalArgumentException: newPosition > limit: (60 > 0)
	at java.base/java.nio.Buffer.createPositionException(Unknown Source)
	at java.base/java.nio.Buffer.position(Unknown Source)
	at java.base/java.nio.ByteBuffer.position(Unknown Source)
	at java.base/java.nio.ByteBuffer.position(Unknown Source)
	at io.netty.buffer.UnsafeByteBufUtil.setBytes(UnsafeByteBufUtil.java:563)
	at io.netty.buffer.PooledUnsafeDirectByteBuf.setBytes(PooledUnsafeDirectByteBuf.java:205)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1113)
	at io.vertx.proton.impl.ProtonTransport.flush(ProtonTransport.java:277)
	at io.vertx.proton.impl.ProtonConnectionImpl.flush(ProtonConnectionImpl.java:296)
	at io.vertx.proton.impl.ProtonDeliveryImpl.flushConnection(ProtonDeliveryImpl.java:137)
	at io.vertx.proton.impl.ProtonDeliveryImpl.settle(ProtonDeliveryImpl.java:131)
	at io.vertx.proton.impl.ProtonDeliveryImpl.disposition(ProtonDeliveryImpl.java:120)
	at io.vertx.proton.ProtonHelper.accepted(ProtonHelper.java:133)
	at org.eclipse.hono.service.amqp.AbstractRequestResponseEndpoint.handleRequestMessage(AbstractRequestResponseEndpoint.java:312)
	at org.eclipse.hono.service.amqp.AbstractRequestResponseEndpoint.lambda$onLinkAttach$1(AbstractRequestResponseEndpoint.java:253)
	at io.vertx.proton.impl.ProtonReceiverImpl.onDelivery(ProtonReceiverImpl.java:236)
	at io.vertx.proton.impl.ProtonTransport.handleSocketBuffer(ProtonTransport.java:168)
	at io.vertx.core.net.impl.NetSocketImpl.lambda$new$1(NetSocketImpl.java:100)
	at io.vertx.core.streams.impl.InboundBuffer.handleEvent(InboundBuffer.java:239)
	at io.vertx.core.streams.impl.InboundBuffer.write(InboundBuffer.java:129)
	at io.vertx.core.net.impl.NetSocketImpl$DataMessageHandler.handle(NetSocketImpl.java:414)
	at io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:55)
	at io.vertx.core.impl.ContextBase.emit(ContextBase.java:239)
	at io.vertx.core.net.impl.NetSocketImpl.handleMessage(NetSocketImpl.java:390)
	at io.vertx.core.net.impl.ConnectionBase.read(ConnectionBase.java:157)
	at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:153)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1383)
	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1246)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1295)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:529)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:468)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:800)
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:499)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:397)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Unknown Source)
```

**And also some NullPointerExceptions:**

```
2023-10-24 14:27:19,511 ERROR [io.qua.ver.cor.run.VertxCoreRecorder] (vert.x-eventloop-thread-1) Uncaught exception received by Vert.x: java.lang.NullPointerException: Cannot invoke "org.apache.qpid.proton.engine.impl.EndpointImpl.setTransportPrev(org.apache.qpid.proton.engine.impl.EndpointImpl)" because the return value of "org.apache.qpid.proton.engine.impl.EndpointImpl.transportNext()" is null
	at org.apache.qpid.proton.engine.impl.ConnectionImpl.removeModified(ConnectionImpl.java:283)
	at org.apache.qpid.proton.engine.impl.EndpointImpl.clearModified(EndpointImpl.java:163)
	at org.apache.qpid.proton.engine.impl.TransportImpl.processEnd(TransportImpl.java:1060)
	at org.apache.qpid.proton.engine.impl.TransportImpl.writeInto(TransportImpl.java:373)
	at org.apache.qpid.proton.engine.impl.TransportOutputAdaptor.pending(TransportOutputAdaptor.java:59)
	at org.apache.qpid.proton.engine.impl.TransportOutputAdaptor.head(TransportOutputAdaptor.java:80)
	at org.apache.qpid.proton.engine.impl.SaslImpl$SwitchingSaslTransportWrapper.head(SaslImpl.java:847)
	at org.apache.qpid.proton.engine.impl.HandshakeSniffingTransportWrapper.head(HandshakeSniffingTransportWrapper.java:151)
	at org.apache.qpid.proton.engine.impl.TransportImpl.outputConsumed(TransportImpl.java:1547)
	at io.vertx.proton.impl.ProtonTransport.flush(ProtonTransport.java:279)
	at io.vertx.proton.impl.ProtonConnectionImpl.flush(ProtonConnectionImpl.java:296)
	at io.vertx.proton.impl.ProtonReceiverImpl.flushConnection(ProtonReceiverImpl.java:175)
	at io.vertx.proton.impl.ProtonReceiverImpl.flow(ProtonReceiverImpl.java:149)
	at io.vertx.proton.impl.ProtonReceiverImpl.flow(ProtonReceiverImpl.java:135)
	at org.eclipse.hono.service.amqp.AbstractRequestResponseEndpoint.flowCreditToRequestor(AbstractRequestResponseEndpoint.java:513)
	at org.eclipse.hono.service.amqp.AbstractRequestResponseEndpoint.lambda$handleRequestMessage$10(AbstractRequestResponseEndpoint.java:357)
	at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Composition$1.onSuccess(Composition.java:62)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Mapping.onSuccess(Mapping.java:40)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Otherwise.onSuccess(Otherwise.java:28)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Composition$1.onSuccess(Composition.java:62)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.SucceededFuture.addListener(SucceededFuture.java:88)
	at io.vertx.core.impl.future.Composition.onSuccess(Composition.java:43)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Composition$1.onSuccess(Composition.java:62)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Composition$1.onSuccess(Composition.java:62)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.SucceededFuture.addListener(SucceededFuture.java:88)
	at io.vertx.core.impl.future.Composition.onSuccess(Composition.java:43)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Composition$1.onSuccess(Composition.java:62)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.SucceededFuture.addListener(SucceededFuture.java:88)
	at io.vertx.core.impl.future.Composition.onSuccess(Composition.java:43)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Mapping.onSuccess(Mapping.java:40)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Otherwise.onSuccess(Otherwise.java:28)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Mapping.onSuccess(Mapping.java:40)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Composition$1.onSuccess(Composition.java:62)
	at io.vertx.core.impl.future.FutureImpl$ListenerArray.onSuccess(FutureImpl.java:262)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.FixedMapping.onSuccess(FixedMapping.java:31)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.Composition$1.onSuccess(Composition.java:62)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.SucceededFuture.addListener(SucceededFuture.java:88)
	at io.vertx.core.impl.future.Composition.onSuccess(Composition.java:43)
	at io.vertx.core.impl.future.FutureImpl$ListenerArray.onSuccess(FutureImpl.java:262)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23)
	at io.vertx.core.Promise.complete(Promise.java:66)
	at org.eclipse.hono.client.pubsub.publisher.PubSubPublisherClientImpl$1.lambda$onSuccess$0(PubSubPublisherClientImpl.java:128)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:264)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:246)
	at io.vertx.core.impl.EventLoopContext.lambda$runOnContext$0(EventLoopContext.java:43)
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:403)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Unknown Source)
```

**How to reproduce:**
On a sandbox cluster it seems to be reproducible if we disconnect a lot of MQTT devices (>500) at once. This way we also see a lot of timeouts during unsubscribing (unregistering the consumer with the command router). Sometimes this even leads to a state in which 1/(Number MQTT Verticles) of devices cannot subscribe (error registering consumer with the command router service) even if they are connected slowly.

Has anyone else ever experienced this behavior or has an idea on how to fix this?
If any additional information is needed I am happy to provide it.

### PR DESCRIPTION
**Pull request by calohmn**
This fixes #3574.

### CODE REVIEW

File: site/homepage/content/release-notes.md [hunk 1]
```diff
@@ -11,6 +11,9 @@ description = "Information about changes in recent Hono releases. Includes new f
 * When running in a Kubernetes cluster with nodes using cgroups v2, the 'hono.command_internal.*' Kafka topics were not
   being cleaned up. This has been fixed. Note that the solution requires the Hono protocol adapter pods to have
   a service account with an assigned RBAC role that allows to perform "get" on the "pods" resource.
+* When using Pub/Sub messaging, there were potentially issues concerning the AMQP connection between protocol adapter
+  and command router, leading for example to timeouts when MQTT devices subscribed/unsubscribed to the command topic.
+  This has been fixed.
```
Comment (mattkaem) [inline site/homepage/content/release-notes.md:16]: FMPOV we should also merge this into the 2.4.0 release. What do you think?

Comment (calohmn) [reply site/homepage/content/release-notes.md:16]: Yes, after merging this PR, I'll apply the change also to the 2.4.x and 2.3.x branches.

Comment (mattkaem) [reply site/homepage/content/release-notes.md:16]: Good, do you than still want to list this as a fix for 2.5.0?

Comment (calohmn) [reply site/homepage/content/release-notes.md:16]: This PR is for the master branch and therefore is for the upcoming 2.5.0 release. 
I've assigned #3574 also to the 2.4.1 and 2.3.2 release projects. I'll apply the change to the corresponding branches directly then, after this PR here got approved and merged.

**Review by mattkaem — APPROVED** [review]
LGTM. Thanks for the PR!

**Pull request merged**